spec/integration/image-streamer/api300/artifact_bundle/create_spec.rb

```diff
@@ -0,0 +1,110 @@
+# (C) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+require 'spec_helper'
+
+klass = OneviewSDK::ImageStreamer::API300::ArtifactBundle
+RSpec.describe klass, integration_i3s: true, type: CREATE, sequence: i3s_seq(klass) do
+  include_context 'integration i3s api300 context'
+
+  let(:artifact_bundle_file_path) { '/tmp/artifact_bundle.zip' }
+  let(:backup_bundle_file_path) { '/tmp/backup_bundle.zip' }
+  let(:deployment_group) { OneviewSDK::ImageStreamer::API300::DeploymentGroup.get_all($client_i3s_300).first }
+  let(:deployment_plan) { OneviewSDK::ImageStreamer::API300::DeploymentPlan.get_all($client_i3s_300).first }
+  let(:plan_script) { OneviewSDK::ImageStreamer::API300::PlanScript.get_all($client_i3s_300).first }
+
+  describe '#create' do
+    it 'should create a artifact bundle' do
+      options = {
+        name: ARTIFACT_BUNDLE1_NAME,
+        description: 'Description of ArtifactBundle'
+      }
+      item = klass.new($client_i3s_300, options)
+      item.add_deployment_plan(deployment_plan, false)
+
+      expect { item.create! }.not_to raise_error
+      expect(item.retrieve!).to eq(true)
+      expect(item['uri']).to be
+      expect(item['name']).to eq(options[:name])
+      expect(item['description']).to eq(options[:description])
+      expect(item['deploymentPlans'].size).to eq(1)
+
+      # creating the read only artifact bundle
+      item = klass.new($client_i3s_300, options)
+      item['name'] = ARTIFACT_BUNDLE2_NAME
+      item.add_plan_script(plan_script)
+      expect { item.create! }.not_to raise_error
+      expect(item['planScripts'].size).to eq(1)
+    end
+  end
+
+  describe '#download' do
+    it 'should download the artifact bundle' do
+      created_item = OneviewSDK::ImageStreamer::API300::ArtifactBundle.find_by($client_i3s_300, name: ARTIFACT_BUNDLE2_NAME).first
+      expect { created_item.download(artifact_bundle_file_path) }.not_to raise_error
+      expect(File.exist?(artifact_bundle_file_path)).to eq(true)
```

Comment:
Do we want to clean up this temp file after we're done with it? I'm not sure if it's the correct solution, but there's a `::Tempfile` class that both takes care of the temp directory (good for Windows and Linux I assume), and deletes it when the related object is garbage-collected.